radix-engine-interface/src/api/kernel_modules/auth_api.rs

```diff
@@ -0,0 +1,8 @@
+use crate::{api::types::*, blueprints::resource::AccessRule};
+use sbor::rust::fmt::Debug;
+
+pub trait ClientAuthApi<E: Debug> {
+    fn get_auth_zone(&mut self) -> Result<ObjectId, E>;
+
+    fn assert_access_rule(&mut self, rule: AccessRule) -> Result<(), E>;
```

Comment:
Now that I see it in code, this method does seem a bit awkward. Why isn't it a method on an auth zone again?

Reply:
Ah is this because, we are asserting access_rule on the previous boundary authzone?

Reply:
Exactly, it's more a method of the stack, rather than an individual authzone.